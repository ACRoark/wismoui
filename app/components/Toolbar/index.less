@import (reference) '../../styles/themes/index.less';

@toolbar-cls-prefix: ~'@{dte-prefix}-toolbar';

.@{toolbar-cls-prefix} {
  background-color: @dte-gray-400;
  padding: 5px;
  text-align: right;
  width: 100%;
}
