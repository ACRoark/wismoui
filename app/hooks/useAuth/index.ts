import { useSelector } from 'react-redux';

import { IApplicationRootState, IAuthenticationState } from 'types';

const useAuth = (): IAuthenticationState =>
  useSelector((state: IApplicationRootState): IAuthenticationState => state.auth);

export default useAuth;
