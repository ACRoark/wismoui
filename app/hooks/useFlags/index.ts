import { useSelector } from 'react-redux';

import { IApplicationRootState, IFeatureFlags } from 'types';

const useFlags = (): IFeatureFlags => useSelector((state: IApplicationRootState): IFeatureFlags => state.config.flags);

export default useFlags;
