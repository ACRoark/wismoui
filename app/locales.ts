export const DEFAULT_LOCALE = 'en';

const PROD_LOCALES = {
  en: 'English',
};

const TEST_LOCALES = {
  en: 'English',
  es: 'Spanish',
};

export const SUPPORTED_LOCALES = process.env.NODE_ENV === 'production' ? PROD_LOCALES : TEST_LOCALES;
