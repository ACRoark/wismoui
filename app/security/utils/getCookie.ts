const getCookie = (name: string): string =>
  document.cookie.split('; ').reduce((previousValue: string, currentValue: string): string => {
    const parts = currentValue.split('=');

    return parts[0] === name ? decodeURIComponent(parts[1]) : previousValue;
  }, '');

export default getCookie;
