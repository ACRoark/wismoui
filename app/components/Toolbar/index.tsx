import React from 'react';

import SelectLanguage from 'components/SelectLanguage';
import useAuthState from 'hooks/useAuth';
import useFlags from 'hooks/useFlags';

import './index.less';

const renderLanguageSelector = (): React.ReactElement => (
  <span key="language" style={{ margin: '0px 25px' }}>
    <SelectLanguage />
  </span>
);

const renderLoginButton = (): React.ReactElement => {
  const onClick = (): void => {
    window.location.href = '/login.html';
  };

  return (
    <button key="login" type="button" onClick={onClick}>
      Login
    </button>
  );
};

const renderLogoutButton = (): React.ReactElement => {
  const onClick = (): void => {
    window.location.href = '/logout.html';
  };

  return (
    <button key="logout" type="button" onClick={onClick}>
      Logout
    </button>
  );
};

const renderItems = (canChangeLanguage: boolean, isAuthenticated: boolean): React.ReactElement[] => {
  const items: React.ReactElement[] = [];

  if (isAuthenticated) {
    items.push(renderLogoutButton());
  } else {
    items.push(renderLoginButton());
  }

  if (canChangeLanguage) {
    items.push(renderLanguageSelector());
  }

  return items;
};

const Toolbar: React.FC = (): React.ReactElement => {
  const { canChangeLanguage, developerMode } = useFlags();

  const { isAuthenticated, loading } = useAuthState();

  if (loading || !developerMode) {
    return <></>;
  }

  return <div className="dte-wismo-toolbar">{renderItems(canChangeLanguage, isAuthenticated)}</div>;
};

export default Toolbar;
