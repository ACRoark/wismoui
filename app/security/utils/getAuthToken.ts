import getCookie from './getCookie';

const getAuthToken = (): string => getCookie('token');

export default getAuthToken;
