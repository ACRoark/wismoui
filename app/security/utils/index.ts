import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
