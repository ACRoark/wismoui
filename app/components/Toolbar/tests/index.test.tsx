import React from 'react';

import { SUPPORTED_LOCALES } from 'locales';

import useAuth from 'hooks/useAuth';
import useFlags from 'hooks/useFlags';
import { createSnapshotWithIntl, shallowWithIntl } from 'testing/utils';

import Toolbar from '..';

jest.mock('hooks/useAuth');
jest.mock('hooks/useFlags');

const mockAuth = useAuth as jest.MockedFunction<typeof useAuth>;
const mockFlags = useFlags as jest.MockedFunction<typeof useFlags>;

const expectToolbarToMatchSnapshot = (isAuthenticated: boolean, language: string, canChangeLanguage: boolean = false): void => {
  mockAuth.mockReturnValue({ isAuthenticated, loading: false });
  mockFlags.mockReturnValue({ canChangeLanguage, developerMode: true });

  const tree = createSnapshotWithIntl(<Toolbar />, language);

  expect(tree).toMatchSnapshot();
};

describe('<Toolbar />', (): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);

  languages.forEach((language: string): void => {
    it(`should not log errors in console when rendered in ${language}`, (): void => {
      const spy = jest.spyOn(global.console, 'error');

      shallowWithIntl(<Toolbar />, language);

      expect(spy).not.toHaveBeenCalled();
    });

    it(`should not render the language selector when changing languages is disabled in the config file (${language})`, (): void =>
      expectToolbarToMatchSnapshot(false, language, false));

    it(`should render the language selector when changing languages is enabled in the config file (${language})`, (): void =>
      expectToolbarToMatchSnapshot(false, language, true));

    it(`should render Login button when the current user is not authenticated (${language})`, (): void =>
      expectToolbarToMatchSnapshot(false, language));

    it(`should render Logout button when the current user is authenticated (${language})`, (): void =>
      expectToolbarToMatchSnapshot(true, language));
  });
});
