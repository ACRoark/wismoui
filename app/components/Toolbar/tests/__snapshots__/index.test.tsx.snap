// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toolbar /> should not render the language selector when changing languages is disabled in the config file (en) 1`] = `
<div
  className="dte-wismo-toolbar"
>
  <button
    onClick={[Function]}
    type="button"
  >
    Login
  </button>
</div>
`;

exports[`<Toolbar /> should not render the language selector when changing languages is disabled in the config file (es) 1`] = `
<div
  className="dte-wismo-toolbar"
>
  <button
    onClick={[Function]}
    type="button"
  >
    Login
  </button>
</div>
`;

exports[`<Toolbar /> should render Login button when the current user is not authenticated (en) 1`] = `
<div
  className="dte-wismo-toolbar"
>
  <button
    onClick={[Function]}
    type="button"
  >
    Login
  </button>
</div>
`;

exports[`<Toolbar /> should render Login button when the current user is not authenticated (es) 1`] = `
<div
  className="dte-wismo-toolbar"
>
  <button
    onClick={[Function]}
    type="button"
  >
    Login
  </button>
</div>
`;

exports[`<Toolbar /> should render Logout button when the current user is authenticated (en) 1`] = `
<div
  className="dte-wismo-toolbar"
>
  <button
    onClick={[Function]}
    type="button"
  >
    Logout
  </button>
</div>
`;

exports[`<Toolbar /> should render Logout button when the current user is authenticated (es) 1`] = `
<div
  className="dte-wismo-toolbar"
>
  <button
    onClick={[Function]}
    type="button"
  >
    Logout
  </button>
</div>
`;

exports[`<Toolbar /> should render the language selector when changing languages is enabled in the config file (en) 1`] = `
<div
  className="dte-wismo-toolbar"
>
  <button
    onClick={[Function]}
    type="button"
  >
    Login
  </button>
  <span
    style={
      Object {
        "margin": "0px 25px",
      }
    }
  >
    <select
      onChange={[Function]}
      value="en"
    >
      <option
        value="en"
      >
        English
      </option>
      <option
        value="es"
      >
        Spanish
      </option>
    </select>
  </span>
</div>
`;

exports[`<Toolbar /> should render the language selector when changing languages is enabled in the config file (es) 1`] = `
<div
  className="dte-wismo-toolbar"
>
  <button
    onClick={[Function]}
    type="button"
  >
    Login
  </button>
  <span
    style={
      Object {
        "margin": "0px 25px",
      }
    }
  >
    <select
      onChange={[Function]}
      value="es"
    >
      <option
        value="en"
      >
        English
      </option>
      <option
        value="es"
      >
        Spanish
      </option>
    </select>
  </span>
</div>
`;
